File: twoPointers/easy/python/validPalindrome.py
import re

# reversing
class Solution:
  def isPalindrome(self, s: str) -> bool:
    new = ''
    for a in s:
      if a.isalpha() or a.isdigit():
        new += a.lower()
    return (new == new[::-1])
  
#two pointers
class Solution:
  def isPalindrome(self, s: str) -> bool:
    s = s.lower()
    pattern = re.compile('[\W_]')
    s = re.sub(pattern, '', s)

    if len(s) == 0:
      return True

    left, right = 0, len(s)-1

    while left < right:
      if s[left] != s[right]:
        return False
      left += 1
      right -= 1

    return True

File: 2-twoPointers/easy/javascript/validPalindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  let clean = s.replace(/[^a-z0-9]/gi, "").toLowerCase();
  if (clean.length === 0) return true;

  let left = 0;
  let right = clean.length - 1;

  for (let i = 0; i < clean.length; i++) {
    if (clean[left] !== clean[right]) return false;

    left++;
    right--;
  }

  return true;
};
